File: DarkOrbit 10.0/Net/netty/commands/KillScreenOptionTypeModule.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Ow.Utils;

namespace Ow.Net.netty.commands
{
    class KillScreenOptionTypeModule
    {
        public const short ID = 11653;

        public const short FREE_PHOENIX = 0;
        public const short BASIC_REPAIR = 1;
        public const short AT_JUMPGATE_REPAIR = 2;
        public const short AT_DEATHLOCATION_REPAIR = 3;
        public const short AT_SECTOR_CONTROL_SPAWNPOINT = 4;
        public const short EXIT_SECTOR_CONTROL = 5;
        public const short BASIC_FULL_REPAIR = 6;
        public const short short_1942 = 7;
        public const short short_564 = 8;
        public const short const_2728 = 9;

        public short typeValue = 0;

        public KillScreenOptionTypeModule(short pRepairTypeValue)
        {
            this.typeValue = pRepairTypeValue;
        }

        public byte[] write()
        {
            var param1 = new ByteArray(ID);
            param1.writeShort(this.typeValue);
            param1.writeShort(20436);
            return param1.Message.ToArray();
        }
    }
}

File: DarkOrbit 9.0/DarkOrbit 9.0 Emulator/src/simulator/netty/serverCommands/KillScreenOptionTypeModule.java
package simulator.netty.serverCommands;

import java.io.DataOutputStream;
import java.io.IOException;

import simulator.netty.ServerCommand;

/**
 Created by Pedro on 30-03-2015.
 */
public class KillScreenOptionTypeModule
        implements ServerCommand {

    public static short FREE_PHOENIX                 = 0;
    public static short BASIC_REPAIR                 = 1;
    public static short AT_JUMPGATE_REPAIR           = 2;
    public static short AT_DEATHLOCATION_REPAIR      = 3;
    public static short AT_SECTOR_CONTROL_SPAWNPOINT = 4;
    public static short EXIT_SECTOR_CONTROL          = 5;
    public static short BASIC_FULL_REPAIR            = 6;
    public static short short_1942                   = 7;
    public static short short_564                    = 8;
    public static short const_2728                   = 9;

    public static int ID = 29900;

    public short mRepairTypeValue = 0;

    public KillScreenOptionTypeModule(short pRepairTypeValue) {
        this.mRepairTypeValue = pRepairTypeValue;
    }

    public void write(DataOutputStream param1) {
        try {
            param1.writeShort(ID);
            this.writeInternal(param1);
        } catch (IOException e) {
        }
    }

    protected void writeInternal(DataOutputStream param1) {
        try {
            param1.writeShort(this.mRepairTypeValue);
            param1.writeShort(17851);
        } catch (IOException e) {
        }
    }
}
